src/Entity/OgRole.php

```diff
@@ -165,4 +200,123 @@ public function save() {
     parent::save();
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function set($property_name, $value) {
+    // Prevent the ID, role type, entity type or bundle to be changed for any of
+    // the default roles. These default roles are required and shouldn't be
+    // tampered with.
+    $is_locked_property = in_array($property_name, [
+      'id',
+      'role_type',
+      'group_type',
+      'group_bundle',
+    ]);
+    $is_default_role = $this->getRoleType() !== self::ROLE_TYPE_STANDARD;
+    if ($is_locked_property && $is_default_role && !$this->isNew()) {
```

Comment:
this check is actually also correct for `ROLE_TYPE_STANDARD` - as those roles shouldn't switch role_types/ group_types/ etc' as-well.

In that case, we might change exception to simple be called `OgRoleException`?